scripts/clean-typescript-files.ps1

```diff
@@ -1,12 +1,28 @@
-$directories = Get-ChildItem -Path $env:MainDirectory  -Directory -Exclude @("graphclient")
-foreach ($directory in $directories) {
-	Remove-Item -Path $directory.FullName -Recurse -Force -Verbose
-}
+param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $targetDirectory,
+    [string]
+    $packageName = "@microsoft/msgraph-sdk-javascript"
+)
+Push-Location $targetDirectory
 
-$files = Get-ChildItem -Path $env:MainDirectory -File  -Filter "*.ts"
-foreach ($file in $files) {
-  Remove-Item -Path $file.FullName -Force
-}
+$kiotaLockFileName = "kiota-lock.json"
+$mainPackageDirectoryName = $packageName.Split("/")[1]
+$mainPackageDirectoryPath = Join-Path $targetDirectory -ChildPath $mainPackageDirectoryName
 
+Push-Location $mainPackageDirectoryPath
+Get-ChildItem -Directory | ForEach-Object { Remove-Item -r $_.FullName }
+Remove-Item $kiotaLockFileName
+Pop-Location
 
-Write-Host "Removed the existing generated files in the repo's main directory: $env:MainDirectory" -ForegroundColor Green
+$directories = Get-ChildItem -Directory -Exclude $mainPackageDirectoryName | Where-Object { -not($_.Name.EndsWith("-tests")) }
+foreach ($directory in $directories) {
+    Push-Location $directory.FullName
+    Get-ChildItem -Directory | ForEach-Object {Remove-Item -r $_.FullName}
+	Remove-Item *.ts -Exclude "index.ts", "*ServiceClient.ts"
```

Comment:
QQ: Is it a manual step to create the `*ServiceClient.ts` and `Readme.md` in the individual packages? (Unfortunately, I can't seem to spot how that is done in the files used in CI steps)
Essentially, it looks like if a new root segment is added to the api, there should be an extra step to add these "config" files for the new package. Correct?

Reply:
that's correct, as designed now whenever a new root segment shows up we'll need to:
- run the create script (creates the package, copies a bunch of files)
- make some manual edits in the package.json (description, etc), Readme.md....
- manually create the specific service client

We could think about automating the creation every time a new root path segment appears, but I didn't think it'd be useful since:
- we'll still have to make some manual edits.
- the API council is working really hard to not introduce new root path segments anymore.